api_controller/models.py

```diff
@@ -0,0 +1,139 @@
+from django.db import models
+from django.conf import settings
+from translation.models import TranslatedLesson, TranslationStep
+from translation.models import YandexTranslator
+
+import requests
+
+
+class ApiController(models.Model):
+    api_key = 0
+    oauth_credentials = {
+        "client_id": settings.STEPIK_CLIENT_ID,
+        "client_secret": settings.STEPIK_CLIENT_SECRET,
+    }
+    base_url = models.TextField()
+    api_version = 0.1
+    api_host = "https://stepik.org"
+    token = None
+
+    def stepik_oauth(self):
+
+        auth = requests.auth.HTTPBasicAuth(self.oauth_credentials["client_id"], self.oauth_credentials["client_secret"])
+        response = requests.post('https://stepik.org/oauth2/token/',
+                                 data={'grant_type': 'client_credentials'},
+                                 auth=auth)
+        self.token = response.json().get('access_token', None)
+        print(self.token)
+        if not self.token:
+            print('Unable to authorize with provided credentials')
+            exit(1)
+
+    def fetch_stepik_object(self, obj_class, obj_id):
+        api_url = '{}/api/{}s/{}'.format(self.api_host, obj_class, obj_id)
+        print("API_URL", api_url)
+        response = requests.get(api_url,
+                                headers={'Authorization': 'Bearer ' + self.token}).json()
+        if '{}s'.format(obj_class) in response:
+            return response['{}s'.format(obj_class)][0]
+        return None
+
+    def fetch_stepik_step(self, pk):
+        return self.fetch_stepik_object("step", pk)
+
+    def fetch_stepik_lesson(self, pk):
+        return self.fetch_stepik_object("lesson", pk)
+
+    def steps_text(self, ids):
+        texts = []
+        for id in ids:
+            step = self.fetch_stepik_object("step", id)
+            if step is None or 'text' in step['block']:
+                texts.append("")
+            else:
+                texts.append(step['block']['text'])
+        return texts
+
+    def create_translation(self, obj_type, pk, service_name=None, lang=None):
+        if self.translation_services.filter(service_name=service_name.lower()).exists():
+            translation_service = self.translation_services.get(service_name=service_name.lower())
+        else:
+            return None
+
+        if obj_type == "steps":
```

Comment:
obj_type was "step" in the code above. Better make named constants (enum) for object types.